### PR TITLE
deprecate Supplier<Response> constructors in HandledTransportAction

### PR DESCRIPTION
**Pull request by talevy**
This commit deprecates all constructors of HandledTransportAction
that take in a Supplier instead of a Writeable.Reader for response
objects.

in addition to the deprecation, the following modules were updated to
leverage Writeable

- modules:ingest-common
- modules:lang-mustache

relates #34389.

### CODE REVIEW

File: modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/GrokProcessorGetAction.java [hunk 1]
```diff
@@ -45,21 +45,25 @@
...
-    @Override
-    public Response newResponse() {
-        return new Response(null);
+        super(NAME, Response::new);
     }
 
     public static class Request extends ActionRequest {
+
+        public Request() {
+            super();
```
Comment (rjernst) [inline modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/GrokProcessorGetAction.java]: explicit empty super is not necessary

File: modules/lang-mustache/src/main/java/org/elasticsearch/script/mustache/MultiSearchTemplateRequest.java [hunk 2]
```diff
@@ -47,6 +47,17 @@
 
     private IndicesOptions indicesOptions = IndicesOptions.strictExpandOpenAndForbidClosedIgnoreThrottled();
 
+    public MultiSearchTemplateRequest() {
+        super();
```
Comment (rjernst) [inline modules/lang-mustache/src/main/java/org/elasticsearch/script/mustache/MultiSearchTemplateRequest.java]: not necessary

**Review by rjernst — APPROVED** [review]
LGTM

**Comment by talevy** [comment]
run elasticsearch-ci/packaging-sample

**Comment by talevy** [comment]
@elasticmachine test this please

**Comment by talevy** [comment]
run elasticsearch-ci/2

**Pull request merged**